### PR TITLE
Feature/sl 961/move hearing to session

### PR DESCRIPTION
**Pull request by radoslawlandowskikainos**

### CODE REVIEW

File: src/app/core/callendar/components/calendar.component.ts [hunk 1]
```diff
@@ -176,4 +185,51 @@ export class CalendarComponent implements OnInit {
 
         eventCallback.emit(event);
     }
+
+    public drop(event) {
+        let selectedSessionId = this.selectedSessionId;
+
+        console.log(`After drop: ${selectedSessionId}`);
```
Comment (r-kruk-kainos) [inline src/app/core/callendar/components/calendar.component.ts]: remove

File: src/app/common/ng-fullcalendar/ng-full-calendar.component.ts [hunk 2]
```diff
@@ -227,6 +230,32 @@ export class NgFullCalendarComponent implements AfterViewInit {
                 elem[i].dispatchEvent(widgetEvent);
             }
         };
+        this.options.drop = (date: any, jsEvent: Event, ui: any, resourceId?: any) => {
+           let detail = { date: date, jsEvent: jsEvent, ui: ui, resourceId: resourceId };
+           const widgetEvent = new CustomEvent('drop', {
+                        bubbles: true,
```
Comment (zielinskikamil) [inline src/app/common/ng-fullcalendar/ng-full-calendar.component.ts]: formatting ;D

File: src/app/core/callendar/components/calendar.component.ts [hunk 3]
```diff
@@ -1,9 +1,15 @@
-import { IcalendarTransformer } from '../transformers/icalendar-transformer';
-import { Component, EventEmitter, Input, OnInit, Output, ViewChild, AfterViewInit } from '@angular/core';
+import { Component, EventEmitter, Input, OnInit, Output, ViewChild } from '@angular/core';
 import * as moment from 'moment';
 import { Default } from 'fullcalendar/View';
-import 'jquery-ui/ui/widgets/draggable.js';
 import { NgFullCalendarComponent } from '../../../common/ng-fullcalendar/ng-full-calendar.component';
+import 'jquery-ui/ui/widgets/draggable.js';
```
Comment (r-kruk-kainos) [inline src/app/core/callendar/components/calendar.component.ts]: It's not needed here

File: src/app/hearing-part/components/draggable-hearing-part/draggable-hearing-part.component.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+})
+export class DraggableHearingPartComponent implements AfterViewChecked {
+    @Input() hearingPart: HearingPart;
+
+    constructor() {
+    }
+
+    ngAfterViewChecked() {
+        ($('.draggable-hearing') as any).draggable({
+            revert: false,
```
Comment (r-kruk-kainos) [inline src/app/hearing-part/components/draggable-hearing-part/draggable-hearing-part.component.ts]: Shouldn't this be true? What will happen if you drop hearingPart on blank space?

File: src/app/core/callendar/components/calendar.component.ts [hunk 5]
```diff
@@ -58,7 +67,7 @@ export class CalendarComponent implements OnInit {
     @Output() eventResizeCallback = new EventEmitter();
     @Output() eventDropCallback = new EventEmitter();
 
-    constructor() {
+    constructor(public hearingModificationService: HearingPartModificationService, public dialog: MatDialog) {
```
Comment (zielinskikamil) [inline src/app/core/callendar/components/calendar.component.ts]: does both these services need to be public?

File: src/app/core/callendar/components/calendar.component.ts [hunk 6]
```diff
@@ -176,4 +185,51 @@ export class CalendarComponent implements OnInit {
...
+                        start: null
+                    } as SessionAssignment);
+                    this.openSummaryDialog();
+                }
+            });
+        }
+    }
+
+    public eventMouseOver(event) {
+        console.log(`Selected: ${event.detail.event.id}`);
```
Comment (zielinskikamil) [inline src/app/core/callendar/components/calendar.component.ts]: plz remove this log too

File: src/app/sessions/components/details-dialog/details-dialog.component.ts [hunk 7]
```diff
@@ -17,6 +18,10 @@ export class DetailsDialogComponent {
         this.dialogRef.close();
     }
 
+    ngAfterViewChecked() {
```
Comment (r-kruk-kainos) [inline src/app/sessions/components/details-dialog/details-dialog.component.ts]: Is this needed?

File: src/app/sessions/effects/session.effects.ts [hunk 8]
```diff
@@ -144,9 +144,9 @@ export class SessionEffects {
     getSessionAfterCommit: Observable<Action> = this.actions$.pipe(
         ofType<sessionTransactionActs.TransactionCommitted>(sessionTransactionActs.EntityTransactionActionTypes.TransactionCommitted),
         withLatestFrom(this.store, (action, state) => state.sessions.sessionTransaction.entities[action.payload].entityId),
-        distinctUntilChanged(),
+       // distinctUntilChanged(),
```
Comment (r-kruk-kainos) [inline src/app/sessions/effects/session.effects.ts]: ?

File: src/app/sessions/models/session-dialog-details.model.ts [hunk 9]
```diff
@@ -1,28 +1,39 @@
 import { SessionViewModel } from './session.viewmodel';
 import * as moment from 'moment';
 import { SessionsStatisticsService } from '../services/sessions-statistics-service';
+import { Observable } from 'rxjs/Observable';
 
 export class SessionDialogDetails {
     private sessionsStatsService: SessionsStatisticsService;
 
-    constructor(public session: SessionViewModel) {
+    public time;
```
Comment (r-kruk-kainos) [inline src/app/sessions/models/session-dialog-details.model.ts]: What about types?

File: src/app/hearing-part/components/draggable-hearing-part/draggable-hearing-part.component.ts [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+            helper: function () { return $(this).clone()
```
Comment (zielinskikamil) [inline src/app/hearing-part/components/draggable-hearing-part/draggable-hearing-part.component.ts]: plz double check this line. I'm pretty sure that `this` there will match function context, not a component class.
If you would like to match component class just change it to `() => { ... `

you can also change formatting, put `{` and `}` in new lines

File: src/app/sessions/components/details-dialog/details-dialog.component.ts (hunk 7 above)
Comment (zielinskikamil) [inline src/app/sessions/components/details-dialog/details-dialog.component.ts]: this method together with `AfterViewChecked` can be removed

File: src/app/sessions/effects/session.effects.ts (hunk 8 above)
Comment (zielinskikamil) [inline src/app/sessions/effects/session.effects.ts]: what with this comment?:)

File: src/app/hearing-part/components/draggable-hearing-part/draggable-hearing-part.component.ts (hunk 4 above)
Comment (radoslawlandowskikainos) [reply src/app/hearing-part/components/draggable-hearing-part/draggable-hearing-part.component.ts]: Don't worry about it

File: src/app/hearing-part/components/draggable-hearing-part/draggable-hearing-part.component.ts (hunk 10 above)
Comment (radoslawlandowskikainos) [reply src/app/hearing-part/components/draggable-hearing-part/draggable-hearing-part.component.ts]: You are 100% percent right, the 'this' will match the function context which in 'helper' function is the currently dragged element and it is what I want it to be.

File: src/app/sessions/models/session-dialog-details.model.ts [hunk 11]
```diff
@@ -1,28 +1,39 @@
...
+import { Observable } from 'rxjs/Observable';
 
 export class SessionDialogDetails {
     private sessionsStatsService: SessionsStatisticsService;
 
-    constructor(public session: SessionViewModel) {
+    public time: any;
+    public endTime: any;
+    public allocatedHearingsDuration: any;
+    public availableDuration: any;
```
Comment (r-kruk-kainos) [inline src/app/sessions/models/session-dialog-details.model.ts]: 🥇

**Pull request merged**